pkg/util/status/build_status.go

```diff
@@ -114,10 +116,36 @@ const (
 )
 
 // NewFailureReason initializes a new failure reason that contains both the
-// reason and a message to be displayed
+// reason and a message to be displayed.
 func NewFailureReason(reason api.StepFailureReason, message api.StepFailureMessage) api.FailureReason {
 	return api.FailureReason{
 		Reason:  reason,
 		Message: message,
 	}
 }
+
+// AddStepInfo appends timing information about specific build stepts to the
+// result object. In case the step was already added, it will update the stop
+// time, this allows timing for substeps to be aggregated under a generic step.
```

Comment:
i'm not sure how i feel about this.  If i run 6 steps and label them all "GatherInput" but they are intermingled with other actions, you're going to report that GatherInput took way longer than it really did.

I think we should just aim to structure the code+definition of "steps" better such that we can explicitly start/stop the timing for a given step exactly once.  If there are cases where you think we are going a little bit of input gathering, then go off and do something else, then come back and do some more input gathering, we should discuss them and decide if maybe we just need more fine grained "step" definitions rather than lumping all those things as "input gathering".  And again i think a lot of this goes away anyway once you stop worrying about timings for flows that are either

1) specific to extended builds
2) specific to standalone s2i usage because openshift handles that step externally